src/test/java/io/github/wimdeblauwe/hsbt/mvc/support/PartialXpathResultMatchers.java

```diff
@@ -0,0 +1,176 @@
+package io.github.wimdeblauwe.hsbt.mvc.support;
+
+import org.hamcrest.Matcher;
+import org.springframework.lang.Nullable;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.util.XpathExpectationsHelper;
+import org.springframework.test.web.servlet.ResultMatcher;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import javax.xml.xpath.XPathExpressionException;
+import java.io.UnsupportedEncodingException;
+import java.util.Map;
+
+/**
+ * Factory for assertions on the response content using XPath expressions when using HtmxPartials
+ *
+ * <p>An instance of this class is typically accessed via
+ * {@link PartialXpathResultMatchers#partialXpath}.
+ *
+ */
+public class PartialXpathResultMatchers {
+
+    private final XpathExpectationsHelper xpathHelper;
+
+    public static PartialXpathResultMatchers partialXpath(String expression, Object... args) throws XPathExpressionException {
+        return new PartialXpathResultMatchers("//htmxPartial"+expression, null, args);
```

Comment:
This helper wraps the response in `<htmxPartial>` and also automatically prepends the xpath query with `//htmxPartial`, making it a well formed document, but keeps the test readable. (the `//htmxPartial/` will appear in assertion failures though.)

Reply:
ok, nice idea!